src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/SqlImportReindexer.cs

```diff
@@ -146,25 +150,37 @@ private async Task ExecuteCommandsWorker(Queue<(string tableName, string indexNa
 
         private async Task ExecuteCommand(string tableName, string indexName, string command, CancellationToken cancellationToken)
         {
-            _logger.LogInformation(string.Format("table: {0}, index: {1} rebuild index start at {2}", tableName, indexName, DateTime.Now.ToString("hh:mm:ss tt")));
+            var retries = 0;
+            var sw = Stopwatch.StartNew();
+            string message;
+        retry:
+            _logger.LogInformation(string.Format("started. table: {0}, index: {1}, retries: {2}", tableName, indexName, retries));
             try
             {
-                using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
-                using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand())
-                {
-                    sqlCommandWrapper.CommandTimeout = _importTaskConfiguration.InfinitySqlTimeoutSec;
-
-                    VLatest.ExecuteCommandForRebuildIndexes.PopulateCommand(sqlCommandWrapper, tableName, indexName, command);
-                    await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
-                }
+                using var sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, false);
+                using var sqlCommandWrapper = sqlConnectionWrapper.CreateRetrySqlCommand();
+                sqlCommandWrapper.CommandTimeout = _importTaskConfiguration.InfinitySqlTimeoutSec;
+                VLatest.ExecuteCommandForRebuildIndexes.PopulateCommand(sqlCommandWrapper, tableName, indexName, command);
+                await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
             }
             catch (SqlException ex)
             {
-                _logger.LogError(ex, "rebuild execute failed");
+                if (ex.IsRetriable())
```

Comment:
Should we have a max amount of retries so that we don't end up in a loop?

Reply:
I don't think so.